doc/adr/2022-11-16-option-precedence-hierarchy.md

```diff
@@ -0,0 +1,28 @@
+**Title**: Establish how multiple values for the same option are reconciled.
+
+**Status**: proposed
+
+**Context**: There are a variety of ways that options can be customized for a
+given rpc call. Both `Client` and `Connection` classes can be created with a set
+of `google::cloud::Options`. Individual function calls in a `Client` can be
+passed `google::cloud::Options`. Some environment variables can be set that
+such that the variable value sets an Option associated with that environment
+variable. Lastly, GCS `Request` types inherit from various Option templates
+which allow the corresponding option to be set in an instance of the `Request`.
+
+When the same Option is set, with different values, via more than one of these
+mechanisms, there needs to be a uniform process of determining which value to
+use.
+
+**Decision**: The following hierarchy will be followed when determining which
+value to use when the same Option is set via multiple mechanisms:
+
+1. Environment variables
+1. Per function call/GCS `Request` instance (if both are set with different values, an error is returned)
+1. `Client` Options
+1. `Connection` Options
```

Comment:
optional: spell these out? It's obvious to me what they mean. But it may not be obvious to someone looking at the ADR with fresh eyes.

1. `*Client::*Client(..., Options)`
2. `Make*Connection(..., Options)`

Reply:
or:

1. `*Client` constructor options
1. `*Connection` factory function options